rubin_sim/scheduler/modelObservatory/model_observatory.py

```diff
@@ -450,16 +450,44 @@ def check_mjd(self, mjd, cloud_skip=20.0):
 
     def _update_rotSkyPos(self, observation):
         """If we have an undefined rotSkyPos, try to fill it out."""
+
+        # Grab the rotator limit from the observatory model
+        rot_limit = [
+            self.observatory.telrot_minpos_rad + 2.0 * np.pi,
+            self.observatory.telrot_maxpos_rad,
+        ]
+
         alt, az = _approx_RaDec2AltAz(
             observation["RA"],
             observation["dec"],
             self.site.latitude_rad,
             self.site.longitude_rad,
             self.mjd,
         )
+
         obs_pa = _approx_altaz2pa(alt, az, self.site.latitude_rad)
-        observation["rotSkyPos"] = (obs_pa + observation["rotTelPos"]) % (2 * np.pi)
-        observation["rotTelPos"] = 0.0
+
+        # If the observation has a rotTelPos set, use it to compute rotSkyPos
+        if (np.isfinite(observation["rotSkyPos"])) | (
```

Comment:
why is np.isfinite(observation["rotSkyPos"]) evaluated twice here? isn't  "x|x" always "x"?